doc/guides/device_mgmt/smp_protocol.rst

```diff
@@ -0,0 +1,289 @@
+    +-------------------+---------------------------------------------------+
+    | ``Seqence Num``   | This is a frame sequence number.                  |
+    |                   | The number is increased by one with each frame    |
+    |                   | request frame.                                    |
+    |                   | Response to a request should copy a sequence      |
+    |                   | number from a request it is responding to.        |
+    |                   |                                                   |
+    +-------------------+---------------------------------------------------+
+    | ``Command ID``    | This is a command, within ``Group``.              |
+    |                   | Response to a request should copy a command ID    |
+    |                   | from a request it is responding to.               |
+    |                   |                                                   |
+    +-------------------+---------------------------------------------------+
+    | ``Data``          | This is data payload of the ``Data Length``       |
+    |                   | size. It is optional as ``Data Length`` may be    |
+    |                   | set to zero, which means that no data follows     |
+    |                   | the header.                                       |
+    |                   |                                                   |
+    +-------------------+---------------------------------------------------+
+
+.. note::
+    Although it may look strange that a frame begins with unused field (``Res``),
+    when the frame gets passed through UART/serial, by default, each byte
+    is sent in LSb to MSb order, which means that ``OP`` bits are sent first.
+
+
+    Contents of a ``Data`` depends on a value of an ``OP``, a ``Group ID``,
+    and a ``Command ID``.
+
+.. _mcumgr_smp_protocol_op_code:
+
+Operation code
+==============
+
+The operation code determines whether an information is written to a device or
+requested from it and whether a packet contains request to a SMP server or
+response from it.
+
+Following operation codes are defined
+
+.. table::
+    :align: center
+
+    +---------------+-----------------------------------------------+
+    | Decimal ID    | Operation                                     |
+    +===============+===============================================+
+    | ``0``         | write request                                 |
+    +---------------+-----------------------------------------------+
+    | ``1``         | write response                                |
+    +---------------+-----------------------------------------------+
+    | ``2``         | read request                                  |
+    +---------------+-----------------------------------------------+
+    | ``3``         | read response                                 |
+    +---------------+-----------------------------------------------+
```

Comment:
Wrong way around, read request is 0, read response is 1, write request is 2, write response is 3

Reply:
Done